projects/knora-ui/src/lib/viewer/values/time-value/time-input/time-input.component.ts

```diff
@@ -0,0 +1,236 @@
+  }
+
+  set placeholder(plh) {
+    this._placeholder = plh;
+    this.stateChanges.next();
+  }
+
+  private _placeholder: string;
+
+  @Input() readonly = false;
+
+  @HostBinding('attr.aria-describedby') describedBy = '';
+
+  setDescribedByIds(ids: string[]) {
+    this.describedBy = ids.join(' ');
+  }
+
+  @Input()
+  get value(): string | null {
+    if (this.form.valid) {
+      const userInput = new DateTime(this.form.value.date, this.form.value.time);
+      return this.userInputToTimestamp(userInput);
+    }
+    return null;
+  }
+
+  set value(timestamp: string | null) {
+    if (timestamp !== null) {
+      const dateTime = this.convertTimestampToDateTime(timestamp);
+      this.form.setValue({date: dateTime.date, time: dateTime.time});
+    } else {
+      this.form.setValue({date: null, time: null});
+    }
+    this.stateChanges.next();
+  }
+
+  @Input() errorStateMatcher: ErrorStateMatcher;
+
+  constructor(fb: FormBuilder,
+              @Optional() @Self() public ngControl: NgControl,
+              private fm: FocusMonitor,
+              private elRef: ElementRef<HTMLElement>,
+              @Optional() _parentForm: NgForm,
+              @Optional() _parentFormGroup: FormGroupDirective,
+              _defaultErrorStateMatcher: ErrorStateMatcher) {
+
+    super(_defaultErrorStateMatcher, _parentForm, _parentFormGroup, ngControl);
+
+    this.dateFormControl = new FormControl(null, [Validators.required]);
+
+    this.timeFormControl = new FormControl(null,  [Validators.required, Validators.pattern(CustomRegex.TIME_REGEX)]);
+
+    this.form = fb.group({
+      date: this.dateFormControl,
+      time: this.timeFormControl
+    });
+
+    fm.monitor(elRef.nativeElement, true).subscribe(origin => {
+      this.focused = !!origin;
+      this.stateChanges.next();
+    });
+
+    if (this.ngControl != null) {
+      this.ngControl.valueAccessor = this;
+    }
+  }
+
+  ngDoCheck() {
+    if (this.ngControl) {
+      this.updateErrorState();
+    }
+  }
+
+  ngOnDestroy() {
+    this.stateChanges.complete();
+  }
+
+  onContainerClick(event: MouseEvent) {
+    if ((event.target as Element).tagName.toLowerCase() != 'input') {
+      this.elRef.nativeElement.querySelector('input').focus();
+    }
+  }
+
+  writeValue(datetime: string | null): void {
+    this.value = datetime;
+  }
+
+  registerOnChange(fn: any): void {
+    this.onChange = fn;
+  }
+
+  registerOnTouched(fn: any): void {
+    this.onTouched = fn;
+  }
+
+  setDisabledState(isDisabled: boolean): void {
+    this.disabled = isDisabled;
+  }
+
+  _handleInput(): void {
+    this.onChange(this.value);
+  }
+
+  // return converted Date obj as a string without the milliseconds
+  userInputToTimestamp(userInput: DateTime): string {
+    const splitTime = userInput.time.split(':');
+    const updateDate = new Date(userInput.date.toCalendarPeriod().periodStart.year,
+                                (userInput.date.toCalendarPeriod().periodStart.month - 1),
```

Comment:
Please add a comment about 0-based indexes for months

Reply:
fixed in 19c3324